Document the format of the field.cattle.io/resourceQuota annotation (#1230)
rancher/rancher-product-docs: versions/v2.15/modules/en/pages/cluster-admin/project-admin/project-resource-quotas/resource-quotas-in-projects.adoc

= How Resource Quotas Work in {rancher-product-name} Projects
:page-languages: [en, de, es, fr, ja, pt, zh]
:revdate: 2026-07-13
:page-revdate: {revdate}
:community-path: how-to-guides/advanced-user-guides/manage-projects/manage-project-resource-quotas/about-project-resource-quotas.adoc 
:product-path: cluster-admin/project-admin/project-resource-quotas/resource-quotas-in-projects.adoc
ifeval::["{build-type}" != "community"]

> :xref-override-default-limit-in-namespaces: xref:cluster-admin/project-admin/project-resource-quotas/override-default-limit-in-namespaces.adoc

Reviewer: ```suggestion
:xref-override-default-limit-in-namespaces: xref:cluster-admin/project-admin/project-resource-quotas/override-default-limit-in-namespaces.adoc
:xref-resource-quota-types: xref:cluster-admin/project-admin/project-resource-quotas/resource-quota-types.adoc
```

endif::[]
ifeval::["{build-type}" == "community"]

> :xref-override-default-limit-in-namespaces: xref:how-to-guides/advanced-user-guides/manage-projects/manage-project-resource-quotas/override-default-limit-in-namespaces.adoc

Reviewer: ```suggestion
:xref-override-default-limit-in-namespaces: xref:how-to-guides/advanced-user-guides/manage-projects/manage-project-resource-quotas/override-default-limit-in-namespaces.adoc
:xref-resource-quota-types: xref:how-to-guides/advanced-user-guides/manage-projects/manage-project-resource-quotas/resource-quota-types.adoc
```

will assign a resource quota that has a *zero* amount for any resource that requested more capacity than what remains in the project.

To create a namespace in an existing project via `kubectl`, use the `field.cattle.io/projectId` annotation. To override the default

> requested quota limit, use the `field.cattle.io/resourceQuota` annotation. This can of course also be {xref-override-default-limit-in-namespaces}[done through the dashboard].

Reviewer: Suggesting the following for this paragraph:

To create a namespace in an existing project using `kubectl`, add the `field.cattle.io/projectId` annotation. To override the default requested quota limit, use the `field.cattle.io/resourceQuota` annotation. You can also configure these settings in the Rancher UI. For more information, see {xref-override-default-limit-in-namespaces}[override default limits in namespaces].

| Applies resource quotas to namespaces through propagation.
| Applies only to the assigned namespace.
|===

== The `field.cattle.io/resourceQuota` annotation

> The `field.cattle.io/resourceQuota` annotation is a JSON object with a single key, `limit`.  The value of this key is a JSON object again, supporting the following keys, each of which maps to a [resource quota type](resource-quota-types.md).  Their values are always strings, except for `extended`, whose value is a JSON object with arbitrary resource references as its keys, and strings as values.

Reviewer: ```suggestion
The `field.cattle.io/resourceQuota` annotation is a JSON object containing a single key, `limit`. The value of `limit` is a JSON object with keys that map to {xref-resource-quota-types}[resource quota types].

All key values are strings, except for `extended`. The `extended` key contains a JSON object that maps arbitrary resource references to string values.
```
